atlp-rwanda/stackup2-barefoot-backend
# #171380755 Email and inapp notification

#### What does this PR do?

- It provides both in-app and Email notifications upon user create trip request

#### Description of Task to be completed?

- When a user creates a trip request they get an email notification about their trip request.

- A user can also disable email notifications so that they don't receive email notifications anymore.

- A manager is able to get the in-app notifications.

#### How should this be manually tested?

- `git clone https://github.com/Stackup-Rwanda/stackup2-barefoot-backend.git`
- `cd stackup2-barefoot-backend`
- `npm install`
- Use `.env.example` as reference for environment variables
- In case you don't have a functional database for this project, run `npm run create-db`
- `npm run migrations`
- `npm run dev`
- At this point, the server must be running and the application should be connected to the database and all the tables in the database are created.
- Signup using this URL `localhost:3000/api/auth/signup` use this sample object 
```
{
	"firstName": "YOUR_FIRST_NAME",
	"lastName":"YOUR_LAST_NAME",
	"address":"YOUR_ADDRESS",
	"username":"YOUR_USERNAME",
	"password":"YOUR_PASSWORD",
	"email":"YOUR_EMAIL",
	"gender":"YOUR_GENDER"
}
```
- Login in order to verify your account using this URL `localhost:3000/api/auth/login` use this sample object

```
"email":"YOUR_EMAIL",
"password":"YOUR_PASSWORD"

```
- create a trip using this URL `localhost:3000/api/trips` use this object sample 
```
{
	"travelFrom": "kigali",
    "travelTo": "Butare",
    "travelDate": "2020-05-14",
    "travelReason": "business meeting",
    "travelType": "one-way",
    "accommodation": true
}

```

- Use this URL to disable email notifications `localhost:3000/api/auth/email/disable/`

- If you are a manager in this app use this URL to get in-app notifications `localhost:3000/api/trips/inapp/notification/`

#### Any background context you want to provide?

- When you signup please provide a real email address that you can access because notifications will be sent that email address that you provided during signup.

#### What are the relevant pivotal tracker stories?

- [171380755](https://www.pivotaltracker.com/story/show/171380755)

#### Screenshots (if appropriate)
![disable email](https://user-images.githubusercontent.com/37122177/77662143-2b3cea00-6f84-11ea-9ed2-23e1f10630a6.PNG)
![email notification](https://user-images.githubusercontent.com/37122177/77662150-2d06ad80-6f84-11ea-9867-f208a62d35dd.PNG)
![in-app](https://user-images.githubusercontent.com/37122177/77662159-2e37da80-6f84-11ea-9219-b90e21ce41d4.PNG)
